flink-runtime/src/main/java/org/apache/flink/runtime/metrics/ReporterSetup.java

```diff
@@ -379,6 +379,7 @@ private static Optional<MetricReporter> loadReporter(
         if (reporterClassName != null) {
             LOG.warn(
                     "The reporter configuration of '{}' configures the reporter class, which is a deprecated approach to configure reporters."
+                            + " The used reporter might not even support reflection-based instantiation, anymore, leading to unexpected errors."
```

Comment:
```suggestion
                            + " The used reporter might not support this configuration, which would result in the reporter not being loaded."
```

- anymore: There may be reporters that never supported this in the first place.
- reflection-base instantiation: relatively technical description
- unexpected errors: they are kinda expected :D

Reply:
I force-pushed a slightly different version. I don't like the `which would result in the reporter not being loaded` since it implies that things might still work (just with the reporter not being available). I would rather point out that errors might be the result of the reporter not supporting the config anymore...